src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -863,7 +869,18 @@ internal bool MethodGroupIsCompatibleWithDelegate(BoundExpression receiverOpt, b
             }
 
             var sourceConstantValue = source.ConstantValue;
-            if (sourceConstantValue == null || sourceConstantValue.IsBad)
+            if (sourceConstantValue == null)
+            {
+                if (conversion.Kind == ConversionKind.DefaultOrNullLiteral && source.Kind == BoundKind.DefaultLiteral)
+                {
+                    return destination.GetDefaultValue();
```

Comment:
This is the existing method used for making a ConstantValue for `default()` operator.
Code is here http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/TypeSymbolExtensions.cs,264ab804a03fe275